setup.json

```diff
@@ -97,12 +97,13 @@
             "notebook~=6.1,>=6.1.5"
         ],
         "pre-commit": [
-            "mypy==0.910",
+            "mypy==0.930",
             "packaging==20.3",
             "pre-commit~=2.2",
             "pylint~=2.11.1",
             "pylint-aiida~=0.1.1",
             "pylint-django",
+            "sqlalchemy[mypy]~=1.4.29",
             "sqlalchemy2-stubs"
```

Comment:
sqlalchemy[mypy] just replaces sqlalchemy2-stubs: https://docs.sqlalchemy.org/en/14/orm/extensions/mypy.html#installation

Reply:
so we remove the `sqlalchemy2-stubs` one?

Reply:
Removed the stubs one. I have to leave now and can't respond again until later tonight

Reply:
yeh heres he exact axtra: https://github.com/sqlalchemy/sqlalchemy/blob/a869dc8fe3cd579ed9bab665d215a6c3e3d8a4ca/setup.cfg#L45-L47,
well I feel the version of sqlalchemy itself should come from the main requirements lock file, so I would remove `sqlalchemy[mypy]~=1.4.29`

Reply:
But then we won't be automatically updated if `sqlachemy` adds another requirement to the `mypy` extra. And only upping the version in our `pre-commit` extra makes sense since that more recent version is only required to make the typing work. It is not necessary for the normal install so would be a bit restrictive to unnecessarily up the minimum requirement there as well, wouldn't it?